service/worker/scheduler/spec.go

```diff
@@ -295,9 +296,7 @@ func (cs *CompiledSpec) CanonicalForm() *schedulepb.ScheduleSpec {
 func (cs *CompiledSpec) GetNextTime(jitterSeed string, after time.Time) GetNextTimeResult {
 	// If we're starting before the schedule's allowed time range, jump up to right before
 	// it (so that we can still return the first second of the range if it happens to match).
-	if cs.spec.StartTime != nil && after.Before(timestamp.TimeValue(cs.spec.StartTime)) {
-		after = cs.spec.StartTime.AsTime().Add(-time.Second)
-	}
+	after = util.MaxTime(after, cs.spec.StartTime.AsTime().Add(-time.Second))
```

Comment:
is there risk of NPE on cs.spec.StartTime?

Reply:
relying on AsTime to return unix epoch, which is way in the past. does it need a comment?